docs/component-specs.md

```diff
@@ -53,6 +53,17 @@ The platforms this input or shipper supports. Must contain one or more of the fo
 
 The output types this input or shipper supports. If this is an input, then inputs of this type can only target (non-shipper) output types in this list. If this is a shipper, then this shipper can only implement output types in this list.
 
+### `proxied_actions` (list of strings)
```

Comment:
Sorry, could you elaborate here a bit more? It's not obvious to me what it would mean to "proxy an action to the corresponding component" -- my impression from looking at the code is that the component is being notified of the action, but I'm not sure what makes it a proxy.

If I was guessing based on the code, I'd call this something like `forwarded_actions` or `notify_actions`. When I see "proxy" I think of something acting as direct intermediary for an action, or masking the identity of the sender/receiver. Is my intuition here right or is "proxy" meant to suggest something else?

Reply:
:-D we had a big discussion on naming here, initial name for this ```agent_actions```, then the agent team asked to changed it to proxied_actions. 
Essentially it is the Agent actions that specific components what to receive. 
For Endpoint tamper protection specifically it needed signed by kibana material (in this case UNENROLL/UPGRADE actions) in order to be able to proceed with unprotecting itself before unenroll or upgrade.

Reply:
Updated the doc. 
Leaving the naming up to @blakerouse and @cmacknz, since the current naming was already changed based on their suggestion.

Reply:
To me the actions are blocking, so they need to be properly handled on the component, before the Elastic Agent can process the action itself. I feel like `forwarded_` means that its just forwarded and not used itself. `proxied` to me means that the action is sent to the component first and then the Elastic Agent does some work itself.

More of a fan of the `proxied_actions`.

Reply:
That's a good point actually, the documentation doesn't mention that these calls can potentially cancel the action and/or block the entire action dispatcher for a significant amount of time, can you add a bit more to make that clear?